docs/sections/user_guide/uw_yaml/rocoto_yaml.rst

```diff
@@ -460,152 +458,166 @@ The ``streq:`` and ``strneq:`` keys compare the values in their ``left:`` and ``
+         state: succeeded
+         task: hello # required
 
   .. code-block:: xml
 
-    <dependency>
-      <taskdep task="hello" state="succeeded" cycle_offset="-06:00:00"/>
-    </dependency>
+     <dependency>
+       <taskdep task="hello" state="succeeded" cycle_offset="-06:00:00"/>
+     </dependency>
 
-* The ``metataskdep:`` key
+* The ``metataskdep:`` key defines a dependency on a metatask:
 
   .. code-block:: yaml
 
-    metataskdep:
-      attrs:
-        cycle_offset: "-06:00:00"
-        state: succeeded
-        metatask: greetings # required
-        threshold: 1
+     metataskdep:
+       attrs:
+         cycle_offset: "-06:00:00"
+         state: succeeded
+         metatask: greetings # required
+         threshold: 1
 
   .. code-block:: xml
 
-    <dependency>
-      <metataskdep metatask="greetings" state="succeeded" cycle_offset="-06:00:00" threshold="1"/>
-    </dependency>
-
-* The ``datadep:`` key
+     <dependency>
+       <metataskdep metatask="greetings" state="succeeded" cycle_offset="-06:00:00" threshold="1"/>
+     </dependency>
 
-  The ``value:`` key for ``datadep:`` accepts a ``cyclestr:`` block.
+* The ``datadep:`` key defines a dependency on on-disk data:
 
   .. code-block:: yaml
 
-    datadep:
-      attrs:
-        age: 120
-        minsize: 1024b
-      value: /path/to/a/file.txt # required
+     datadep:
+       attrs:
+         age: 120
+         minsize: 1024b
+       value: /path/to/a/file.txt # required
 
   .. code-block:: xml
 
-    <dependency>
-      <datadep age="120" minsize="1024b">/path/to/a/file.txt</datadep>
-    </dependency>
+     <dependency>
+       <datadep age="120" minsize="1024b">/path/to/a/file.txt</datadep>
+     </dependency>
 
-* The ``timedep:`` key
+   * The ``value:`` key accepts a ``cyclestr:`` block.
 
-  The ``timedep:`` key will almost certainly want a ``cyclestr:`` block.
+* The ``timedep:`` key defines a dependency on a real-world time:
 
   .. code-block:: text
 
-    timedep:
-      cyclestr:
-        value: @Y@m@d@H@M@S
+     timedep:
+       cyclestr:
+         value: @Y@m@d@H@M@S
 
   .. code-block:: xml
 
-    <dependency>
-      <timedep><cyclestr>@Y@m@d@H@M@S</cyclestr></timedep>
-    </dependency>
+     <dependency>
+       <timedep><cyclestr>@Y@m@d@H@M@S</cyclestr></timedep>
+     </dependency>
+
+  * The ``timedep:`` key will almost certainly want a ``cyclestr:`` block.
+
+* The ``sh:`` key defines a dependency on the successful execution of a shell command:
+
+  .. code-block:: yaml
+
+     sh:
+       command: test $(find /some/dir -type f -name "*.grib2" | wc -l) -eq 24
+
+  .. code-block:: xml
+
+     <dependency>
+       <sh>test $(find /some/dir -type f -name "*.grib2" | wc -l) -eq 24</sh>
+     </dependency>
+
+  * The ``command:`` key accepts a ``cyclestr:`` block.
+  * The ``sh:`` key may be suffixed with ``_<name>`` to provide a unique name for the dependency, e.g. ``sh_foo:`` would translate to XML tag ``<sh name="foo">``.
```

Comment:
The use of `_<name>` here is confusing. I _think_ it's meant to specify that you could fill in any arbitrary name, but looks a lot like XML syntax.

Reply:
Yes, good point, will change. It's meant to follow the *nix standard of `[optional]` vs `<required>` arguments, but is out of place here.